packages/slice-machine/src/modules/pushSaga.ts

```diff
@@ -0,0 +1,149 @@
+import { ComponentUI } from "../../lib/models/common/ComponentUI";
+import { CustomTypeSM } from "@slicemachine/core/build/models/CustomType";
+import {
+  getState,
+  pushCustomType,
+  pushSliceApiClient,
+} from "../../src/apiClient";
+import {
+  all,
+  call,
+  fork,
+  put,
+  SagaReturnType,
+  takeLatest,
+} from "redux-saga/effects";
+import { createAction, getType } from "typesafe-actions";
+import { withLoader } from "./loading";
+import { LoadingKeysEnum } from "./loading/types";
+import { pushCustomTypeCreator } from "./selectedCustomType";
+import { pushSliceCreator } from "./selectedSlice/actions";
+import { openToasterCreator, ToasterType } from "./toaster";
+import { modalOpenCreator } from "./modal";
+import { ModalKeysEnum } from "./modal/types";
+import axios from "axios";
+import { refreshStateCreator } from "./environment";
+
+export const changesPushCreator = createAction("PUSH_CHANGES")<{
+  unSyncedSlices: ReadonlyArray<ComponentUI>;
+  unSyncedCustomTypes: ReadonlyArray<CustomTypeSM>;
+}>();
+
+function* pushSliceSaga(slice: ComponentUI) {
+  try {
+    yield call(pushSliceApiClient, slice);
+    yield put(pushSliceCreator.success({ component: slice }));
+    return undefined;
+  } catch (e) {
+    if (axios.isAxiosError(e) && e.response?.status) {
+      return e.response.status;
+    }
+    return slice.model.id;
+  }
+}
+
+function* pushCustomTypeSaga(customType: CustomTypeSM) {
+  try {
+    yield call(pushCustomType, customType.id);
+    yield put(pushCustomTypeCreator.success({ customTypeId: customType.id }));
+    return undefined;
+  } catch (e) {
+    if (axios.isAxiosError(e) && e.response?.status) {
+      return e.response.status;
+    }
+    return customType.id;
+  }
+}
+
+function isTruthyString(str: unknown): str is string {
+  return typeof str === typeof "" && Boolean(str);
+}
+
+function isAuthErrorCode(n: unknown): n is number {
+  return typeof n === typeof 0 && (n === 401 || n === 403);
+}
+
+export function* changesPushSaga({
+  payload,
+}: ReturnType<typeof changesPushCreator>) {
+  const { unSyncedSlices, unSyncedCustomTypes } = payload;
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const slicesPushResults: Array<string | undefined | number> = yield all(
```

Comment:
What's the type in case of an error ?

Reply:
depends on the error, server error number, anyother error undefined